src/main/java/com/basho/riak/client/core/query/timeseries/TableDefinition.java

```diff
@@ -0,0 +1,163 @@
+package com.basho.riak.client.core.query.timeseries;
+
+import java.util.*;
+
+/**
+ * Holds a definition for a Table in Time Series Riak.
+ * Immutable once created.
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.4
+ */
+public class TableDefinition
+{
+    private final String tableName;
+    private final LinkedHashMap<String, FullColumnDescription> fullColumnDescriptions = new LinkedHashMap<>();
+    private final ArrayList<FullColumnDescription> partitionKeys = new ArrayList<>();
```

Comment:
Not sure that we need to pre-cache keys stuff, since not all users need to use it.

IMO, it is better to introduce utilitarian assessors which will return newly created collections each time. As a result caching will be performed on the user side, if the user really needs this.